queries.sql

```diff
@@ -97,3 +97,103 @@ SELECT species, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight FROM
+SELECT a.name FROM animals a JOIN owners o ON a.owner_id = o.id WHERE o.full_name = 'Dean Winchester' AND a.escape_attempts = 0;
+
+-- Who owns the most animals?
+SELECT o.full_name, COUNT(a.*) AS animal_count FROM owners o LEFT JOIN animals a ON o.id = a.owner_id GROUP BY o.full_name ORDER BY animal_count DESC LIMIT 1;
+
+
+-------------------------
+
+-- Find the last animal seen by William Tatcher
+SELECT a.name AS last_animal_seen_by_William_Tatcher
+FROM visits v
+JOIN vets vet ON v.vet_id = vet.id
+JOIN animals a ON v.animal_id = a.id
+WHERE vet.name = 'Vet William Tatcher'
+ORDER BY v.visit_date DESC
+LIMIT 1;
+
+-- Count the number of different animals seen by Stephanie Mendez
+SELECT COUNT(DISTINCT v.animal_id) AS animals_seen_by_Stephanie_Mendez
+FROM visits v
+JOIN vets vet ON v.vet_id = vet.id
+WHERE vet.name = 'Vet Stephanie Mendez';
+
+-- List all vets and their specialties, including those with no specialties
+SELECT vet.name, COALESCE(species.name, 'No Specialty') AS specialty
+FROM vets vet
+LEFT JOIN specializations spec ON vet.id = spec.vet_id
+LEFT JOIN species ON spec.species_id = species.id;
+
+-- List all animals that visited Stephanie Mendez in a specified date range
+SELECT a.name AS animal_name, v.visit_date
+FROM visits v
+JOIN vets vet ON v.vet_id = vet.id
+JOIN animals a ON v.animal_id = a.id
+WHERE vet.name = 'Vet Stephanie Mendez'
+AND v.visit_date BETWEEN '2020-04-01' AND '2020-08-30';
+
+-- Find the animal with the most visits to vets
+SELECT a.name AS animal_name, COUNT(v.animal_id) AS visit_count
+FROM visits v
+JOIN animals a ON v.animal_id = a.id
+GROUP BY a.name
+ORDER BY visit_count DESC
+LIMIT 1;
+
+-- Find the first animal visited by Maisy Smith
+SELECT a.name AS first_visit_animal
+FROM visits v
+JOIN vets vet ON v.vet_id = vet.id
+JOIN animals a ON v.animal_id = a.id
+WHERE vet.name = 'Vet Maisy Smith'
+ORDER BY v.visit_date ASC
+LIMIT 1;
+
+-- Find details of the most recent visit, including animal, vet, and date
+SELECT a.name AS animal_name, vet.name AS vet_name, v.visit_date
+FROM visits v
+JOIN vets vet ON v.vet_id = vet.id
+JOIN animals a ON v.animal_id = a.id
+ORDER BY v.visit_date DESC
+LIMIT 1;
+
+-- Count the number of visits with vets not specialized in the animal's species
+SELECT COUNT(*) AS visits_with_unspecialized_vet
+FROM visits v
+JOIN vets vet ON v.vet_id = vet.id
+JOIN animals a ON v.animal_id = a.id
+LEFT JOIN specializations spec ON vet.id = spec.vet_id AND spec.species_id = a.species_id
+WHERE spec.species_id IS NULL;
+
+-- Find the specialty that Maisy Smith should consider based on the species she sees the most
+SELECT s.name AS suggested_specialty
+FROM visits v
+JOIN vets vet ON v.vet_id = vet.id
+JOIN animals a ON v.animal_id = a.id
+JOIN specializations spec ON vet.id = spec.vet_id
+JOIN species s ON spec.species_id = s.id
+WHERE vet.name = 'Vet Maisy Smith'
+GROUP BY s.name
+ORDER BY COUNT(*) DESC
+LIMIT 1;
+
```

Comment:
- [ ] Hello! 👋 Great job on using this query to help Maisy Smith find a specialty. 😊 However, I noticed that the query result came back with zero rows, which is not what we expected. 😕 We need to fix this issue. What steps can you take to resolve it? Let's work together to get the right result for Maisy. 🚀👩‍⚕️💼
![Screenshot from 2023-09-30 04-05-07](https://github.com/ishimwepati/vet_clinic_db/assets/112261237/f2da881c-6925-495f-bf87-edfe47df680a)

Reply:
Hello !

Below are the implemented features :

1. First, I need to count the number of visits for each species that "Vet Maisy Smith" has seen. And I achieve this :
```
SELECT a.species_id, COUNT(*) AS visit_count
FROM visits v
INNER JOIN animals a ON v.animal_id = a.id
INNER JOIN vets vt ON v.vet_id = vt.id
WHERE vt.name = 'Vet Maisy Smith'
GROUP BY a.species_id
ORDER BY visit_count DESC
LIMIT 1

```

2. I used the result of the subquery to find the corresponding specialty name from the species table :
```
JOIN species s ON most_visited_species.species_id = s.id
```

3. Theb In the main query, I select the name column from the species table with the alias suggested_specialty.
```
SELECT s.name AS suggested_specialty
```

4. Out here is the result :
<img width="655" alt="Screenshot 2023-09-30 at 11 28 45" src="https://github.com/ishimwepati/vet_clinic_db/assets/14818338/bccf8221-ee98-427c-8329-cf8910df5dad">


Your suggestions are most welcome !